src/src_cryptography/src_kmc_crypto_service/cryptography_interface_kmc_crypto_service.template.c

```diff
@@ -1925,6 +1964,85 @@ int32_t curl_response_error_check(CURL* curl_handle, char* response)
 
 }
 
+int32_t curl_perform_with_cam_retries(CURL* curl_handle,char* response)
+{
+    int32_t status = CRYPTO_LIB_SUCCESS;
+    uint8_t cam_retry = 0;
+    while(cam_retry < CAM_MAX_AUTH_RETRIES)
+    {
+        CURLcode res;
+        res = curl_easy_perform(curl);
+
+        if(res != CURLE_OK) // This is not a response w/return code, this is something breaking!
+        {
+            status = CRYPTOGRAHPY_KMC_CRYPTO_SERVICE_GENERIC_FAILURE;
+            fprintf(stderr, "curl_easy_perform() failed: %s\n",
+                    curl_easy_strerror(res));
+            break; // Go to Post retry loop cleanup and return status.
+        }
+
+        status = curl_response_error_check(curl_handle, response);
+
+        if(status == CRYPTO_LIB_SUCCESS) // Crypto Service REST call worked! Break out of retry loop.
+        {
+            break;
+        }
+
+        if(status == CAM_AUTHENTICATION_REQUIRED) // CAM_AUTHENTICATION_REQUIRED code indicates CAM config setup for authentication
+        {
+#ifdef DEBUG
+            printf("Attempting to authenticate and retrieve CAM SSO Token.\n");
+#endif
+            status = get_cam_sso_token();
+        } else // Conditions not met for CAM retry! Break out of retry loop.
+        {
+            break;
+        }
+        cam_retry++;
+    }
+    return status;
+}
+
+/**
+ * @brief Function: initialize_kerberos_keytab_file_login
+ *
+ *  This function launches the kinit kerberos utility to initialize credentials for the current user.
+ *  It may be possible to do this with the KRB5 API, but for now all we are doing is launching the kinit utility.
+ *  Consider using the KRB5 APIs in a future release: https://web.mit.edu/kerberos/krb5-devel/doc/appdev/refs/api/index.html
+ *
+ *
+**/
+int32_t initialize_kerberos_keytab_file_login(void)
+{
+    int32_t status = CRYPTO_LIB_SUCCESS;
+
+    if(cam_config->keytab_file_path == NULL)
+    {
+        status = CAM_INVALID_CONFIGURATION_KEYTAB_FILE_PATH_NULL;
+        return status;
+    }
+
+    if(cam_config->username == NULL)
+    {
+        status = CAM_INVALID_CONFIGURATION_KEYTAB_FILE_USERNAME_NULL;
+        return status;
+    }
+
+    // Build the kinit shell command with keytab file path + username
+    char* kinit_shell_command_base = "kinit -kt %s %s";
+    uint32_t len_kinit_shell_command = strlen(kinit_shell_command_base) + strlen(cam_config->keytab_file_path) + strlen(cam_config->username);
+    char* kinit_shell_command = malloc(len_kinit_shell_command) + 1;
```

Comment:
Are these freed somewhere that I'm missing?

Reply:
Good catch! I've added the corollary free for this malloc.